docs/concepts/storage.md

```diff
@@ -0,0 +1,92 @@
+
+## How does this align incentives?
+
+Storage-staked tokens are unavailable for other uses such as validation staking. This increases the yield that validators will receive. Learn more in [the economics whitepaper](https://near.org/papers/economics-in-sharded-blockchain/).
+
+## When do tokens get staked?
+
+On each incoming transaction that adds data.
+
+Let's walk through an example:
+
+1. You launch a [a custom token](https://examples.near.org/?q=fungible), deploying your token's smart contract to the account `mycoiniscoolerthanyours.near`
+2. Token holders can name an escrow – an account which can transfer tokens on their behalf. Naming an escrow will require a call to your smart contract, and your smart contract will need to store this mapping of token-owner-to-escrow-account.
+3. When this call comes in, NEAR will check that `mycoiniscoolerthanyours.near` has a large enough balance that it can stake an amount corresponding to this data addition. If it does not, the transaction will fail. Otherwise, these new tokens get locked for storage staking and the transaction completes.
+
+Note in this example that it is the token holders and not your smart contract account which, [by default](https://docs.near.org/docs/concepts/gas#what-about-prepaid-gas), pay the gas fee.
+
+## The "million worthless escrows" attack
+
+Note that this can create an attack surface. To continue the example above, if naming an escrow costs a token holder close to nothing while costing the contract owner significantly more, then a malicious user can exploit the imbalance to make maintaining the contract prohibitively expensive.
+
+Take care, then, when designing your smart contracts to ensure that such attacks cost potential attackers more than it would be worth.
+
+## btw, you can remove data to unstake some tokens
+
+People familiar with the "immutable data" narrative about blockchains find this surprising. While it's true that an _indexing node_ will keep all data forever, _validating nodes_ (that is, the nodes run by most validators in the network) do not. Smart contracts can provide ways to delete data, and this data will be purged from most nodes in the network within a few [epochs](https://docs.near.org/docs/concepts/epoch).
+
+Note that a call to your smart contract to remove data has an associated gas fee. Also, since deletions takes time, there is an upper limit on the amount of data you can remove in a single transaction, as your transaction must be able to fit in the one-second block time.
```

Comment:
…in a one-second block time and due to computational limits per block. (don't know the source)

Reply:
block time is not really one second. It is better to talk about gas limit here.

Reply:
@bowenwang1996 is the gas limit and the reason for it documented somewhere such as Nomicon?